base/strings/string.jl

```diff
@@ -29,16 +29,28 @@ const ByteArray = Union{CodeUnits{UInt8,String}, Vector{UInt8},Vector{Int8}, Fas
 
 @inline between(b::T, lo::T, hi::T) where {T<:Integer} = (lo ≤ b) & (b ≤ hi)
 
+"""
+    String <: AbstractString
+
+The generic, default string type in Julia, used by e.g. string literals.
+
+`String`s are immutable sequences of `Char`s. A `String` is stored internally as
+a contiguous byte array, and while they are interpreted as being UTF-8 encoded,
+they can be composed of any byte sequence.
+This representation often makes `String` appropriate for passing strings to C.
```

Comment:
I understand that this is copied from the old documentation. But, if it can contain any bytes, including null in the middle, I don't think C compatibility is worth mentioning.

```suggestion
```

Reply:
Yes, that's a good point. Also, for interop with `C` one should supposedly use `Cstring` which does not allow null.